pyqir-parser/pyqir/parser/_parser.py

```diff
@@ -605,7 +605,7 @@ def type(self) -> QirType:
         """
         if self._type == None:
             self._type = QirType(self.instr.type)
-        return self._type
+        return cast(QirType, self._type)
```

Comment:
I wonder if we can use [https://docs.python.org/3/library/functools.html#functools.cached_property](`@functools.cached_property`) instead? That would remove the extra cache fields and the casts.